### PR TITLE
Jmprieur/update msal3 0 4 and adal 5 x

### PR DESCRIPTION
**Pull request by jmprieur**
## Purpose

update the sample to msal3 0 4 and adal 5 x

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

```
Follow the README.md
run the TestAll.bat
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace CommonCacheADALV5
+{
+ /// <summary>
+ /// Simple file based persistent cache implementation for a desktop application (from ADAL 4.x)
+ /// </summary>
+ class FilesBasedTokenCache : TokenCache
```
Comment (bgavrilMS) [inline TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs]: Spacing looks a bit off.

File: TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+   // if the access operation resulted in a cache update
+   if (this.HasStateChanged)
+   {
+    lock (FileLock)
+    {
+     // reflect changes in the persistent store
+     WriteToFileIfNotNull(AdalV3CacheFilePath, SerializeAdalV3());
+     WriteToFileIfNotNull(UnifiedCacheFilePath, SerializeMsalV3());
+
+     // once the write operation took place, restore the HasStateChanged bit to false
```
Comment (bgavrilMS) [inline TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs]: does this compile? You don't need to set this flag.

**Review by bgavrilMS — APPROVED** [review]

File: TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs (hunk 2 above)
Comment (jmprieur) [reply TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs]: this does compile. This is ADAL here, @bgavrilMS !

File: TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs (hunk 1 above)
Comment (jmprieur) [reply TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs]: yes. thanks for catching this. I used a 1 space indent to be able to have code that is not too wide for the docs / wikis. Will fix this

File: TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs (hunk 2 above)
Comment (bgavrilMS) [reply TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs]: Ok :). It still does nothing, but we've been showing this pattern everywhere...

Comment (jmprieur) [reply TokenCacheMigration/CommonCacheAdalV5/FilesBasedTokenCache.cs]: it does nothing in ADAL as well?

**Review by jennyf19 — COMMENTED** [review]
LGTM

**Pull request merged**